### PR TITLE
Fix #856: Only cache failed scenario's for rerun

### LINKED ISSUE #856: Rerun should only cache failed scenarios
Rerun in Behat 3 cache scenarios which are failed and skipped both. Scenarios which are skipped were not intended for run and should not be cached for rerun (similar to behat 2.x)

Please consider caching scenarios for rerun, which are failed only.

### PR DESCRIPTION
**Pull request by rajeshtaneja**

### CODE REVIEW

**Comment by everzet** [comment]
Tests are failing.

**Comment by rajeshtaneja** [comment]
grrr.. Sorry about that. Fixed it.

Thanks @everzet

**Comment by OskarStark** [comment]
Fixes #856

File: src/Behat/Behat/Tester/Cli/RerunController.php [hunk 1]
```diff
@@ -113,7 +114,7 @@ public function collectFailedScenario(AfterScenarioTested $event)
             return;
         }
 
-        if ($event->getTestResult()->isPassed()) {
+        if ($event->getTestResult()->getResultCode() != TestResult::FAILED) {
```
Comment (stof) [inline src/Behat/Behat/Tester/Cli/RerunController.php]: please use a strict comparison

**Comment by stof** [comment]
would be great to add a scenario covering this to prevent regressions

**Comment by rajeshtaneja** [comment]
Thanks @stof,

I will try get to sort this in coming week. Sorry quite busy with regular stuff...

**Pull request merged**